File: app/src/main/java/com/robotnec/budget/core/calculator/eval/Evaluator.java
package com.robotnec.budget.core.calculator.eval;

import com.robotnec.budget.core.exception.InvalidExpressionException;

/**
 * @author zak <zak@swingpulse.com>
 */
public interface Evaluator {
    double eval(String expression) throws InvalidExpressionException;
}

File: app/src/main/java/com/robotnec/budget/core/calculator/eval/Evaluator.kt
package com.robotnec.budget.core.calculator.eval

import com.robotnec.budget.core.exception.InvalidExpressionException

/**
 * @author zak zak@swingpulse.com>
 */
interface Evaluator {
    @Throws(InvalidExpressionException::class)
    fun eval(expression: String): String
}
